plugins/modules/files/iso_get_files.py

```diff
@@ -0,0 +1,164 @@
+requirements:
+- "pycdlib"
+- "python >= 2.7"
+version_added: '5.5.0'
+
+options:
+   iso:
+     description:
+     - This is the absolute paths of source ISO file.
+     - Will fail if specified ISO file does not exist on local machine.
+     - 'Note: With all ISO9660 levels from 1 to 3, all file names are restricted to uppercase letters, numbers and
+       underscores (_). File names are limited to 31 characters, directory nesting is limited to 8 levels, and path
+       names are limited to 255 characters.'
+     type: path
+     required: yes
+   get_files:
+     description:
+     - The absolute path with file name on ISO file.
+     type: list
+     elements: dict
+     required: yes
+     suboptions:
+       file_in_iso:
+         description:
+         - the path of file in ISO
+         type: path
+       file_local:
+         description:
+         - the local path
+         type: path
+'''
+
+EXAMPLES = r'''
+- name: "Get file from ISO"
+  community.general.iso_get_files:
+    iso: "/Users/zouy/Documents/ubuntu-22.04-desktop-amd64.iso"
+    get_files:
+      - file_in_iso: "/preseed/ubuntu.seed"
+        file_local: "/tmp/ubuntu.seed.org"
+      - file_in_iso: "/boot/grub/grub.cfg"
+        file_local: "/tmp/grub.cfg.org"
+'''
+
+RETURN = r'''
+iso:
+    description: paths of source ISO file..
+    returned: on success
+    type: str
+    sample: "/path/to/file.iso"
+file_in_iso:
+    description: path of the file in ISO.
+    returned: on success
+    type: str
+    sample: "/preseed/ubuntu.seed"
+file_local:
+    description: path of the local file
+    returned: on success
+    type: str
+    sample: "/tmp/ubuntu.seed"
+
+'''
+
+import os
+import traceback
+
+
+PYCDLIB_IMP_ERR = None
+try:
+    import pycdlib
+    HAS_PYCDLIB = True
+except ImportError:
+    PYCDLIB_IMP_ERR = traceback.format_exc()
+    HAS_PYCDLIB = False
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.common.text.converters import to_native
+
+
+def iso_get_file(module, iso_path, get_files_list):
+    try:
+        iso = pycdlib.PyCdlib()
+        iso.open(iso_path)
+
+        for item in get_files_list:
+            file_in_iso = item['file_in_iso'].strip()
+            file_local = item['file_local'].strip()
+
+            file_local_dir = os.path.dirname(file_local)
+            if not os.path.exists(file_local_dir):
+                try:
+                    os.makedirs(file_local_dir)
+                except OSError as err:
+                    iso.close()
+                    msg = "Failed to create folder %s with error: %s" % (
+                        file_local_dir, to_native(err))
+                    return -1, msg
+
+            record = iso.get_record(rr_path=file_in_iso)
+            if record.is_file():
+                iso.get_file_from_iso(file_local, rr_path=file_in_iso)
+            else:
+                iso.close()
+                return -1, "%s is not a file in ISO" % file_in_iso
```

Comment:
This could be an option for the user: non-existent files should cause an error or should they be silently ignored?

Reply:
Thanks for your code review.
 We have checked the file exists when invoke iso.get_file_from_iso(file_local, rr_path=file_in_iso) :
 record = iso.get_record(rr_path=file_in_iso)
 if record.is_file():
       iso.get_file_from_iso(file_local, rr_path=file_in_iso)